packages/graphql/src/translate/field-aggregations/aggregation-sub-queries.ts

```diff
@@ -17,43 +17,91 @@
  * limitations under the License.
  */
 
-import type { AggregationAuth } from "./field-aggregations-auth";
 import { wrapApocConvertDate } from "../projection/elements/create-datetime-element";
 import { stringifyObject } from "../utils/stringify-object";
+import Cypher from "@neo4j/cypher-builder";
+import { dedent } from "graphql-compose";
 
-export function createMatchWherePattern(matchPattern: string, auth: AggregationAuth, whereInput: string): string {
-    const whereQuery = whereInput || auth.whereQuery ? "WHERE" : "";
-    const andQuery = whereInput && auth.whereQuery ? "AND" : "";
-
-    return `MATCH ${matchPattern} ${whereQuery} ${whereInput} ${andQuery} ${auth.whereQuery}`;
+export function createMatchWherePattern(
+    matchPattern: Cypher.Relationship,
+    directed: boolean,
+    preComputedWhereFields: Cypher.CompositeClause | undefined,
+    auth: Cypher.Predicate | undefined,
+    wherePredicate: Cypher.Predicate | undefined
+): Cypher.Clause {
+    const matchClause = new Cypher.Match(matchPattern.pattern({ directed }));
+    const whereClause = preComputedWhereFields && !preComputedWhereFields?.empty ? new Cypher.With("*") : matchClause;
+    if (wherePredicate) whereClause.where(wherePredicate);
+    if (auth) whereClause.where(auth);
+    return preComputedWhereFields && !preComputedWhereFields?.empty
+        ? Cypher.concat(matchClause, preComputedWhereFields, whereClause)
+        : matchClause;
 }
 
-export function stringAggregationQuery(matchWherePattern: string, fieldName: string, targetAlias: string): string {
-    const fieldPath = `${targetAlias}.${fieldName}`;
-    return `${matchWherePattern}
-        WITH ${targetAlias} as ${targetAlias}
-        ORDER BY size(${fieldPath}) DESC
-        WITH collect(${fieldPath}) as list
-        RETURN {longest: head(list), shortest: last(list)}`;
+export function stringAggregationQuery(
+    matchWherePattern: Cypher.Clause,
+    fieldName: string,
+    fieldRef: Cypher.Variable,
+    targetAlias: Cypher.Node | Cypher.Relationship
+): Cypher.RawCypher {
+    const fieldPath = targetAlias.property(fieldName);
+    return new Cypher.RawCypher((env) => {
+        const targetAliasCypher = targetAlias.getCypher(env);
+        const fieldPathCypher = fieldPath.getCypher(env);
+
+        return dedent`${matchWherePattern.getCypher(env)}
+        WITH ${targetAliasCypher} as ${targetAliasCypher}
+        ORDER BY size(${fieldPathCypher}) DESC
+        WITH collect(${fieldPathCypher}) as list
+        RETURN {longest: head(list), shortest: last(list)} AS ${fieldRef.getCypher(env)}`;
```

Comment:
One style comment and one general question - we don't only return the aggregations from the selection here? We always return `longest` and `shortest` even if only one was asked for?

```suggestion
        RETURN { longest: head(list), shortest: last(list) } AS ${fieldRef.getCypher(env)}`;
```